packages/core/botpress/src/util.js

```diff
@@ -138,6 +141,44 @@ const getCircularReplacer = () => {
 
 const safeStringify = o => JSON.stringify(o, getCircularReplacer())
 
+const validateBotVersion = (bpVersion, botfileVersion) => {
+  if (botfileVersion == null) {
+    throw new Error("Version doesn't exist in botfile.js")
+  }
+
+  if (_.isEmpty(botfileVersion) || !_.isString(botfileVersion)) {
+    throw new Error('Version in botfile.js must be non-empty string specifying the valid semver.')
+  }
+
+  try {
+    // TODO: change this method if "semver" module will implement semver.isValid()
+    semver.valid(botfileVersion)
+  } catch (err) {
+    throw new Error('Version in botfile.js must have proper semver format (e.g. 10.25.0)')
+  }
+
+  const msgPreamble = `Your bot may be incompatible with botpress v${bpVersion}
+  because it looks like it was originally created with botpress v${botfileVersion}.
+  To address this `
+
+  if (semver.lt(bpVersion, botfileVersion)) {
+    throw new Error(
+      msgPreamble +
+        `update the versions of botpress and any @botpress/* modules in your package.json to ${botfileVersion}`
+    )
+  }
+
+  const botfileMajorVersion = Number(semver.major(botfileVersion))
+  const bpMajorVersion = Number(semver.major(bpVersion))
+
+  if (bpMajorVersion > botfileMajorVersion) {
+    throw new Error(
+      msgPreamble +
+        `check https://github.com/botpress/botpress/blob/master/CHANGELOG.md and update your bot for any breaking changes listed there, then update the version in your botfile.js to ${bpVersion}`
```

Comment:
This one seems to be too long

Reply:
It should wrap properly in the terminal, no?

Reply:
It just doesn't look nice in my editor, but I think it's fine. )

Reply:
Ah, could be switched to multiple strings with concatenation